daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/RunnerMain.scala

```diff
@@ -84,8 +84,15 @@ object RunnerMain {
           fileContent.parseJson
         })
 
+        val script = try {
+          Script.fromDar(dar, scriptId)
+        } catch {
+          case e: Throwable =>
+            system.terminate
+            sys.exit(1)
+        }
         val runner = try {
-          new Runner(dar, applicationId, commandUpdater, timeProvider)
+          new Runner(dar, script.scriptIds, applicationId, commandUpdater, timeProvider)
```

Comment:
Assuming I understand your plans for the follow-up PRs correctly, in the end the constructor shouldn’t throw anymore so we can probably remove the try here. Nothing to do in this PR though, just a note for the future :slightly_smiling_face:

Reply:
Agreed.